### PR TITLE
Wind direction generation / tweaks of windpower

### PR DESCRIPTION
**Pull request by ghost**
#### Summary
```SUMMARY: Content "Generates wind direction and improves wind strength random-ness'"```

#### Purpose of change
Currently Wind Strength is generated quite simply : ```W = std::max( 0, 1020 - static_cast<int>( P ) );```
Literally the difference between a number and the current pressure.
This produces some strange results.
I've also added wind direction generation based on real-life data from Boston and surrounding New England area for prevailing winds.
Nothing uses this yet, but sailboats, hot air balloons, sheltering from wind, aerodynamics etc could all potentially use this.

I've made use of data from websites like : 

https://www.windfinder.com/windstatistics/boston_logan_airport
https://weatherspark.com/y/26197/Average-Weather-in-Boston-Massachusetts-United-States-Year-Round

#### Describe the solution
This is currently the wind strength measured per hour from the game code, in Winter.

It goes over 90mph... and it's very random, there is none of the continuity you would see from weather fronts.

image

After the change this is Winter wind speed per hour.

image

You can see there are peaks ( much lower - as per real data ) the seasonal averages match up quite well to real life seasonal averages measured per hour.
The peaks are indicative of generally windy days, they rise and subside, weakly correlated with air pressure in the area.

Conversely, in-game the daily ( per minute ) measurements had no noise to them, because they followed a pressure value, 

Here is in-game day ( per minute )

image

after the change it is more like this : 

image

you can see it is moving towards a new value , with noise, on the way.

The wind direction describes the direction it is blowing from and has a raw angle ( for converting to vectors for vehicles etc ) and a string that prints out from the weather reader CBM ( or for a GUI element )

the distribution has random-ness but a typical winter might look like this : 

image

Throughout the year, the prevailing wind is West to East, though Easterly winds also are quite common, there are seasonal variations, but Northerly and Southerly winds are less common.

#### Describe alternatives you've considered
Tried many different things.

#### Additional context
As with my previous PRs, I'm using this as a C++ learning exercise, so I may have committed some cardinal sins. Apologies if so.
The weather test code currently uses a RNG to set the seed, I used the same seed for all tests however.

For comparison , here is the Summer wind graph - more dead-wind days, solar power would be better here instead of wind turbines!

image

### CODE REVIEW

File: src/weather_gen.cpp [hunk 1]
```diff
@@ -18,35 +24,35 @@ constexpr double tau = 2 * PI;
...
 
 weather_generator::weather_generator() = default;
+int weather_generator::current_winddir = 1000;
 
 w_point weather_generator::get_weather( const tripoint &location, const time_point &t,
                                         unsigned seed ) const
 {
     const double x( location.x /
-                    2000.0 ); // Integer x position / widening factor of the Perlin function.
+                    5000.0 ); // Integer x position / widening factor of the Perlin function.
```
[the remark this answers is not in the document]
Comment (davidpwbrown) [reply src/weather_gen.cpp]: Indeed, that's a big boo-boo, I can just adjust the relevant wind and pressure x,y,z, divisor individually.. have no intention of touching temperature or humidity or acid rain.
Pressure does need the smoothing out though. Adding the commit now.

File: src/weather.cpp [hunk 2]
```diff
@@ -652,6 +652,14 @@ std::string print_temperature( double fahrenheit, int decimals )
     }
 }
 
+std::string print_winddirection( int angle )
+{
+    std::ostringstream ret;
+    ret << std::fixed;
```
Comment (OrenAudeles) [inline src/weather.cpp]: `std::fixed` is unnecessary here as it's a floatfield format flag, no effect.

Also, we know `angle` should be within range `[0, 359]` (3 char) so it's probably faster to just use string here. `std::ostringstream` does a lot of work for such a simple operation.
```
// Using std::string only
std::string print_winddirection( int angle )
{
    assert( angle >= 0 && angle < 360 && "Angle out of bounds [0, 359]" ); // Assert on bad input
    // Ensure size, angle is in range [0, 359], 3 characters + null terminator
    std::string ret( 4, '\0' ); // Might get small string optimization and not heap allocate
    snprintf( &ret.front(), 4, "%d", angle );
    return string_format( pgettext( "wind direction angle", "%s" ), ret.c_str() );
}
```
Or using a char buffer
```
// Using char buffer
std::string print_winddirection( int angle )
{
    assert( angle >= 0 && angle < 360 && "Angle out of bounds [0, 359]" ); // Assert on bad input
    char buf[4] = ""; // Stack allocation of buffer
    snprintf( buff, 4, "%d", angle );
    return string_format( pgettext( "wind direction angle", "%s" ), buf );
}
```
Both alternatives do significantly less work than using `std::ostringstream`.

Comment (davidpwbrown) [reply src/weather.cpp]: I think that entire function is not necessary now, it's a leftover from when I was trying different ways to print the direction string.  The CBM and weather reader iuse just do ```add_msg_if_player( m_info, _( "Wind Direction: From the %s." ), weatherPoint.dirstring );``` 
I can just remove that function? or can you think of a reason to keep it there?

Comment (OrenAudeles) [reply src/weather.cpp]: No idea if it's useful to keep around, just noticed that it wasn't actually used anywhere haha. Using dirstring is probably better, unless the player has a way to tell the exact direction an 8-way discrete representation works

**Pull request merged**